extensions/spring-cloud-config-client/runtime/src/test/java/io/quarkus/spring/cloud/config/client/runtime/SpringCloudConfigClientGatewayTest.java

```diff
@@ -45,26 +47,32 @@ void testBasicExchange() throws Exception {
         wireMockServer.stubFor(WireMock.get(springCloudConfigUrl).willReturn(WireMock
                 .okJson(getJsonStringForApplicationAndProfile(applicationName, profile))));
 
-        final Response response = sut.exchange(applicationName, profile);
 
-        assertThat(response).isNotNull().satisfies(r -> {
-            assertThat(r.getName()).isEqualTo("foo");
-            assertThat(r.getProfiles()).containsExactly("dev");
-            assertThat(r.getPropertySources()).hasSize(4);
-            assertThat(r.getPropertySources().get(0)).satisfies(ps -> {
-                assertThat(ps.getSource()).contains(entry("bar", "spam"), entry("foo", "from foo development"),
-                        entry("democonfigclient.message", "hello from dev profile"));
-            });
-            assertThat(r.getPropertySources().get(1)).satisfies(ps -> {
-                assertThat(ps.getSource()).contains(entry("my.prop", "from application-dev.yml"));
-            });
-            assertThat(r.getPropertySources().get(2)).satisfies(ps -> {
-                assertThat(ps.getSource()).contains(entry("foo", "from foo props"),
-                        entry("democonfigclient.message", "hello spring io"));
-            });
-            assertThat(r.getPropertySources().get(3)).satisfies(ps -> {
-                assertThat(ps.getSource()).contains(entry("foo", "baz"));
-            });
+        sut.exchange(applicationName, profile).invoke(new Consumer<Response>() {
```

Comment:
I think it's best to use `.await()` and then just assert the result as was done in the existing code